docs/sources/flow/reference/cli/tools.md

```diff
@@ -0,0 +1,80 @@
+---
+title: grafana-agent tools
+weight: 100
+---
+
+# `grafana-agent tools` command
+
+The `grafana-agent tools` command contains command line tooling grouped by Flow
+component.
+
+> **NOTE**: utilities in this command have no backwards compatibility
+> guarantees and may change or be removed between releases.
+
+## Subcommands
+
+### prometheus.remote_write sample-stats
+
+Usage: `grafana-agent tools prometheus.remote_write sample-stats [FLAG ...] WAL_DIRECTORY`
+
+The `sample-stats` command reads the Write-Ahead Log (WAL) specified by
+`WAL_DIRECTORY` and collects information on metric samples within it.
+
+For each metric discovered, `sample-stats` emits:
+
+* The timestamp of the oldest sample received for that metric.
+* The timestamp of the newest sample received for that metric.
+* The total number of samples discovered for that metric.
+
+By default, `sample-stats` will return information for every metric in the WAL.
+The `--selector` flag can optionally be passed to filter the reported metrics
+to a smaller set.
+
+The following flags are supported:
+
+* `--selector`: A PromQL label selector to filter data by. (default `{}`)
+
+### prometheus.remote_write target-stats
+
+Usage: `grafana-agent tools prometheus.remote_write target-stats [FLAG ...] WAL_DIRECTORY`
+
+The `target-stats` command reads the Write-Ahead Log (WAL) specified by
+`WAL_DIRECTORY` and collects metric cardinality information for a specific
+target.
+
+For the target specified by the `--job` and `--instance` flags, unique metric
+names for that target are printed along with the number of series with that
+metric name.
+
+The following flags are supported:
+
+* `--job`: The `job` label of the target.
+* `--instance`: The `instance` label of the target.
+
+The `--job` and `--instance` labels are required.
```

Comment:
If these are required should they be defined in the `Usage:` above for copy pastability?